examples/chat/client/chat.cc

```diff
@@ -224,55 +173,43 @@ class Room {
 int main(int argc, char** argv) {
   absl::ParseCommandLine(argc, argv);
 
-  std::unique_ptr<oak::ManagerClient> manager_client;
-  std::string addr = absl::GetFlag(FLAGS_app_address);
-  std::string application_id;
-  std::unique_ptr<OakApplication> application;
+  std::string address = absl::GetFlag(FLAGS_address);
+  LOG(INFO) << "Connecting to Oak Application: " << address;
 
-  if (addr.empty()) {
-    // Connect to the Oak Manager and create the Oak Application.
-    manager_client = absl::make_unique<oak::ManagerClient>(grpc::CreateChannel(
-        absl::GetFlag(FLAGS_manager_address), grpc::InsecureChannelCredentials()));
-    application =
-        absl::make_unique<OakApplication>(manager_client.get(), absl::GetFlag(FLAGS_module));
-    application_id = application->Id();
-    addr = application->Addr();
-    LOG(INFO) << "Connecting to new Oak Application id=" << application_id << " at " << addr;
-  } else {
-    LOG(INFO) << "Connecting to existing Oak Application at " << addr;
-  }
+  oak::ApplicationClient::InitializeAssertionAuthorities();
 
   // Connect to the Oak Application.
   // TODO(#488): Use the token provided on command line for authorization and labelling of data.
-  oak::ApplicationClient::InitializeAssertionAuthorities();
-  auto stub = Chat::NewStub(oak::ApplicationClient::CreateChannel(addr));
+  auto stub = Chat::NewStub(oak::ApplicationClient::CreateChannel(address));
   if (stub == nullptr) {
     LOG(QFATAL) << "Failed to create application stub";
   }
 
-  RoomId room_id;
-  if (!absl::Base64Unescape(absl::GetFlag(FLAGS_room_id), &room_id)) {
-    LOG(QFATAL) << "Failed to parse --room_id as base 64";
-  }
-  std::unique_ptr<Room> room;
-  if (room_id.empty()) {
-    room = absl::make_unique<Room>(stub.get());
-    room_id = room->Id();
-    LOG(INFO) << "Join this room with --app_address=" << addr
-              << " --room_id=" << absl::Base64Escape(room_id);
-  }
+  if (!absl::GetFlag(FLAGS_no_main_loop)) {
```

Comment:
Also, I don't really see what this has to do with the main description of the PR. Just talked to @tiziano88  about this... again, no strong feelings, but ideally it would be nice to have these separated.

Reply:
I also find this flag name confusing, both because of the negation, but also because of the mention of "main loop", which is not very clear. How about a `--test` flag instead? We could also add it to the other examples if necessary if we wanted to make them do something in a non-interactive way as part of the test.

Reply:
Have moved this logic to a separate PR: https://github.com/project-oak/oak/pull/600